homeassistant/components/keymitt_ble/switch.py

```diff
@@ -0,0 +1,48 @@
+"""Switch platform for MicroBot."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+
+from .const import DOMAIN
+from .entity import MicroBotEntity
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up MicroBot based on a config entry."""
+    coordinator: MicroBotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([MicroBotBinarySwitch(coordinator, entry)])
+
+
+class MicroBotBinarySwitch(MicroBotEntity, SwitchEntity):
+    """MicroBot switch class."""
+
+    _attr_has_entity_name = True
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
+        if not self.coordinator.api.is_connected():
```

Comment:
We shouldn't hold the connection forever. It should only connect when you want to take an action as Bluetooth adapters have a limited amount of connections that can be open and keeping the connection open can drain the battery for battery powered devices.  Also if the device gets disconnected you'll have a new failure mode if you don't check every time before executing an action if it's connected.  We usually implement a asyncio.Lock wrapped connect and disconnect check before taking an action.  Then set a timer to disconnect after a few seconds.  There is an example in pySwitchbot.

Reply:
Also, 2022.9 will support multiple adapters so its important they you keep the BLEDevice up to date with the library since it has the routing information needed to connect with the nearest ble controller

<img width="397" alt="Screen Shot 2022-08-18 at 8 28 22 AM" src="https://user-images.githubusercontent.com/663432/185470204-8ba1ac30-f70b-4770-9808-64af7c4eb6b4.png">